### PR TITLE
Change Headers to use spell-case for React Testing and React Ecosystem Sections 

### PR DESCRIPTION
**Pull request by Babb2000**
## Because
Changed Headers to use spell-case

## This PR
- Apply's spell-case to headers

## Issue
Related to #25861

## Pull Request Requirements

-   [x] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
-   [x] The title of this PR follows the `location of change: brief description of change` format, e.g. `Intro to HTML and CSS lesson: Fix link text`
-   [x] The `Because` section summarizes the reason for this PR
-   [x] The `This PR` section has a bullet point list describing the changes in this PR
-   [x] If this PR addresses an open issue, it is linked in the `Issue` section
-   [x] If any lesson files are included in this PR, they have been previewed with the [Markdown preview tool](https://www.theodinproject.com/lessons/preview) to ensure it is formatted correctly
-   [x] If any lesson files are included in this PR, they follow the [Layout Style Guide](https://github.com/TheOdinProject/curriculum/blob/main/LAYOUT_STYLE_GUIDE.md)

### CODE REVIEW

**Comment by ManonLef** [comment]
Hi there! I noticed that at least these lessons are not included in these PR, but they still have a few headings that need to be sentence-cased. Would you mind PR'ing those so I can review them all at once?

- Type Checking With PropTypes
- Styling React Applications (Only the Component Libraries header I believe)

**Comment by ManonLef** [comment]
@Babb2000 just give me an @ ping once all the lessons I mentioned are ready for review okay?

**Comment by ManonLef** [comment]
Hi @Babb2000 ! I'd love to review this but would you mind changing this header still 
- Styling React Applications (Only the Component Libraries header)

**Comment by Babb2000** [comment]
@ManonLef Done

File: react/the_react_ecosystem/react_router.md [hunk 1]
```diff
@@ -12,7 +12,7 @@ This section contains a general overview of topics that you will learn in this l
 - How do you add a "catch-all" route?
 - How do you add protected routes?
 
-### Client-Side Routing
+### Client-Side routing
```
Comment (ManonLef) [inline react/the_react_ecosystem/react_router.md:15]:
```suggestion
### Client-side routing
```

File: react/the_react_ecosystem/react_router.md [hunk 2]
```diff
@@ -144,7 +144,7 @@ export default App;
 
 And now, we don't get the browser reloading every time we click the link on the navbar!
 
-### Nested Routes, Outlets And Dynamic Segments
+### Nested routes, outlets And dynamic segments
```
Comment (ManonLef) [inline react/the_react_ecosystem/react_router.md:147]:
```suggestion
### Nested routes, outlets and dynamic segments
```

File: react/the_react_ecosystem/type_checking_with_proptypes.md [hunk 3]
```diff
@@ -71,7 +71,7 @@ export default RenderName;
 
 In this example, with the help of the defaultProps property we are defining a default value for the `name` prop. This way, if the `RenderName` component is called without passing in the `name` prop, it will default to "Zach". When you do pass in props, they will take precedence over the default props.
 
-### What about TypeScript?
+### What about typeScript?
```
Comment (ManonLef) [inline react/the_react_ecosystem/type_checking_with_proptypes.md:74]:
```suggestion
### What about TypeScript?
```

**Review by ManonLef — CHANGES_REQUESTED** [review]
Thanks for doing this. 
I edited a few as suggestions. 

Instead of locally editing these, you can also click `commit suggestion` on each of them to automatically commit the suggestions here.

File: react/the_react_ecosystem/react_router.md (hunk 1 above)
Comment (ManonLef) [reply react/the_react_ecosystem/react_router.md:15]: this one is not changed yet @Babb2000 :)

**Review by ManonLef — APPROVED** [review]

File: react/the_react_ecosystem/react_router.md (hunk 1 above)
Comment (ManonLef) [reply react/the_react_ecosystem/react_router.md]: I committed this one myself to start wrapping up the issue

**Comment by ManonLef** [comment]
Thanks again!

**Pull request merged**